rustls/src/suites.rs

```diff
@@ -216,6 +213,31 @@ pub enum ConnectionTrafficSecrets {
     },
 }
 
+impl ConnectionTrafficSecrets {
```

Comment:
This feels a little bit awkward, but I also thought having `slice_to_array` and `slices_to_arrays` defined in three places wasn't going to cut it. Having the duplication across two crate-internal places was already a bit crummy, adding a third for the provider example is one too many for sure.

Reply:
I think this is a reasonable approach.